src/mesh.rs

```diff
@@ -107,11 +107,23 @@ impl Hash for DynamicMesh {
 
 impl Mesh {
     /// Set mesh material.
-    pub fn set_material(
+    pub fn set_material<M: Into<Material>>(
         &mut self,
-        material: Material,
+        material: M,
     ) {
-        let msg = Operation::SetMaterial(material);
+        let msg = Operation::SetMaterial(material.into());
+        let _ = self.object.tx.send((self.object.node.downgrade(), msg));
+    }
+
+    /// Enable or disable instanced rendering for this `Mesh`.
+    ///
+    /// It will improve performance in case of multiple `Mesh`es with the same
+    /// materials and geometries.
+    pub fn use_instancing(
```

Comment:
I'd like us to not have this explicit switch, preferably. We can technically draw everything with instancing. Whatever happens to never be instanced will just have an instance of 1. Whatever does get instanced with `instance_mesh` should be automatically fast.